Storage/Storage/Bookmarks.swift

```diff
@@ -34,14 +36,11 @@ public struct BookmarkRoots {
 /**
  * The immutable base interface for bookmarks and folders.
  */
-@objc public protocol BookmarkNode {
-    var guid: String { get }
-    var title: String { get }
-    var icon: UIImage { get }
-}
-
-@objc public protocol Bookmark : BookmarkNode {
-    var url: String! { get }
+public class BookmarkNode {
+    public var id: Int? = nil
+    public var guid: String = ""
+    public var title: String = ""
```

Comment:
Again, a bit legacy. BookmarkNodes technically don't need a url (for folders). I think we started with the assumption that you have to assign a title though i.e. non-null. I needed an ID here in the original patch to do linking against the icons (but I don't even really need that in this patch anymore).